### DISCUSSION TITLE
Hot restart after Flutter and Rust code changes

### DISCUSSION THREAD

**Discussion by xBLACKICEx** [fzyzcjy/flutter_rust_bridge#2465] [Show and tell]
I read the quickstart docs and don't quite get it.

I need to re-run the flutter run every time I change the rust code?

because after I start the `flutter_rust_bridge_codegen generate --watch` command and add the `hello` function, and than hot-restart I get the following error:
![img](https://github.com/user-attachments/assets/51421ba2-a377-462d-9028-322745de5d8e)

If just modifying the existing rust code like changing the greet() function by one letter, nothing changes after hot-restart.

This is confusing to me.

**Comment by fzyzcjy** [comment]
> I need to re-run the flutter run every time I change the rust code?

Currently still yes, but after https://github.com/dart-lang/sdk/issues/55850 is done, we will no longer need that :)
